ui/src/components/dataset/ingestion/IngestionStepper.vue

```diff
@@ -331,86 +345,113 @@ const datasetTypes = [
-  [STEP_KEYS.GENERAL_INFO]: null,
-  [STEP_KEYS.INGEST]: null,
+// `stepIsPristine` determines whether any of the fields in the current step have been interacted with by the user.
+const stepIsPristine = computed(() => {
+  return !!Object.values(stepPristineStates.value[step.value])[0];
 });
 
+const loadingResources = ref(false); // determines if the initial resources needed for the stepper are being fetched
+const searchingFiles = ref(false);
+const validatingForm = ref(false);
 const loading = computed(() => {
   return loadingResources.value || searchingFiles.value || validatingForm.value;
 });
 
-const searchSpaceBasePath = computed(() => searchSpace.value.base_path);
+// Various values related to the submission process.
+const isSubmissionAlertVisible = ref(false);
+const submitAttempted = ref(false);
+const submissionButtonText = ref(SUBMIT_BUTTON.INGEST);
+const submissionSuccess = ref(false);
 
-const _searchText = computed(() => {
-  return (
-    (searchSpace.value.base_path.endsWith("/")
-      ? searchSpace.value.base_path
-      : searchSpace.value.base_path + "/") + fileListSearchText.value
-  );
-});
+/**
+ * Name of the Dataset that the user will ingest. This is set by the user before initiating the ingestion.
+ */
+const ingestedDatasetName = ref("");
+
+// Current step index
+const step = ref(0);
 
 const isLastStep = computed(() => {
   return step.value === steps.length - 1;
 });
 
-const isNextButtonDisabled = computed(() => {
-  return stepHasErrors.value || submissionSuccess.value || loading.value;
-});
+// The file selected by the user for ingestion.
+const selectedFile = ref(null);
 
-const isPreviousButtonDisabled = computed(() => {
-  return step.value === 0 || submissionSuccess.value || loading.value;
+// The list of available Instruments for assigning to the Dataset being ingested.
+const sourceInstrumentOptions = ref([]);
+
+// Stores information about the ingested Dataset.
+const createdDataset = ref(null);
+// The Raw Data that will be assigned to the Dataset being ingested.
+const selectedRawData = ref(null);
+// The (existing) Project that will be assigned to the Dataset being ingested.
+const projectSelected = ref(null);
+// The (new) Project that will be assigned to the Dataset being ingested.
+const projectCreated = ref(null);
+// The Instrument that will be assigned to the Dataset being ingested.
+const selectedSourceInstrument = ref(null);
+
+// determines whether there are any Raw Data options to choose from
+const noRawDataToAssign = ref(false);
+// determines whether there are any Project options to choose from
+const noProjectsToAssign = ref(false);
+// determines whether there are any Instrument options to choose from
+const noInstrumentsToAssign = ref(false);
+
+// Determines whether the Dataset being ingested is of type Raw Data or some other type.
+const willIngestRawData = computed(() => {
+  return (
+    selectedDatasetType.value["value"] === config.dataset.types.RAW_DATA.key
+  );
```

Comment:
JS changes until this point are moving around existing code for better readability, and adding comments. Can be ignored